### ISSUE TITLE
Forum icons supporting FontAwesome

### ISSUE THREAD

**Issue by Moc** [e107inc/e107#3733]
Forum icons currently allow uploading of images but apparently no support yet for fontawesome icons (or glyph).. Could this be added?

**Comment by CaMer0n** [comment]
It's labelled as an icon, but it is actually an image: 

```
'forum_image'   =>   array ( 'title' => LAN_ICON, 'type' => 'image', 'batch'=>false, 'inline'=>false, 'noedit'=>false, 'data' => 'str', 'width' => 'auto', 'help' => 'Will be displayed next to the forum name', 'readParms' => '', 'writeParms' => 'media=forum&max=1', 'class' => 'center', 'thclass' => 'center',  ),

```

Changing type from 'image' to 'icon' would no doubt make some people unhappy. 
We'd need to add another field with type=icon. (and modify the upgrade routines accordingly)

**Comment by Moc** [comment]
@CaMer0n I've done some testing on it and I think we can add this feature without having to add another field and without needing to modify the update routines. 

- If I change `'writeParms' => 'media=forum&max=1',` to 'writeParms' => 'media=forum&max=1&glyph=1',` the Glyph tab shows up and selecting the glyph icon works fine. 
- It saves in the database as (for example) `glyphicon-book
.glyph`
- For a test, I've modified the forum shortcode to include this:
```php
// Check if it's glyph / icon instead of image 
if (strpos($this->var['forum_image'], '.glyph') !== false) 
{
    return e107::getParser()->toIcon($this->var['forum_image'], $parms);
}
```

- This renders correctly as `<span class='glyphicon glyphicon-book'><!-- --></span>`

My issue is that I want to use FontAwesome rather than Glyphicons (especially since Bootstrap no longer includes them as of BS4) 

Ideally, I'd like to have a FontAwesome tab in the media manager modal but I realise this may be quite some work. In the `glyphTab()` method however, I already do some preparations for FontAwesome (next issue is FA4 vs FA5...). 

What are your thoughts on this? I could try and give creating a new method `FaTab()` a go if you like....

**Comment by CaMer0n** [comment]
@Moc Great! That's even better. `glyph=1` should work for FontAwesome too. 
Also, `toIcon()` should automatically detect the `.glyph` extension. 
Or, you could use `toGlyph()` if you already know it's a `.glyph`. 

Re: FontAwesome in MediaManager.. 
Is it possible you need to scroll a few more pages before you see them? (note, I'm on slide 4 here)

![image](https://user-images.githubusercontent.com/2825679/58910518-97372480-86ca-11e9-9387-39dc4f5f502e.png)

**Comment by Moc** [comment]
@CaMer0n I'm going to do a *facepalm* on the mediamanager FA icons..... I am not sure how I missed that haha, thank you! 

I'll work on the needed edits as discussed above :)

**Comment by Moc** [comment]
After toying around with this and after having discussed this with Cameron, we decided to go for adding a new field, rather than to (ab)use the forum image field for this. It would require many tweaks which in fact is adding a mess and is also a reinvention of the wheel (since the Admin UI already provides for an 'icon' type). 

See latest changes. Closing the issue, but let me know if there are any issues arising from this change.
